File: kaggle_environments/envs/kore_fleets/starter_bots/java/test/ObservationTest.java
package test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

import org.junit.Assert;
import org.junit.Test;

import kore.Observation;

public class ObservationTest {
    
    @Test
    public void givenValidObservation_createSuccessful() throws IOException {
        Path observation = Paths.get("bin", "test", "observation.json");
        String rawObservation = Files.readString(observation);        
        
        Observation ob = new Observation(rawObservation);

        Assert.assertEquals(0, ob.player);
        Assert.assertEquals(16, ob.step);
        Assert.assertEquals(4, ob.playerHlt.length);
        Assert.assertEquals(4, ob.playerFleets.size());
        Assert.assertEquals(0, ob.playerFleets.get(0).size());
        Assert.assertEquals(4, ob.playerShipyards.size());
        Assert.assertEquals(1, ob.playerShipyards.get(0).size());
    }

    @Test
    public void givenFullObservation_createSuccessful() throws IOException {
        Path observation = Paths.get("bin", "test", "fullob.json");
        String rawObservation = Files.readString(observation);        
        
        Observation ob = new Observation(rawObservation);

        Assert.assertEquals(0, ob.player);
        Assert.assertEquals(200, ob.step);
        Assert.assertEquals(2, ob.playerHlt.length);
        Assert.assertEquals(2, ob.playerFleets.size());
        Assert.assertEquals(1, ob.playerFleets.get(0).size());
        Assert.assertEquals(2, ob.playerShipyards.size());
        Assert.assertEquals(6, ob.playerShipyards.get(0).size());
    }
}

File: kaggle_environments/envs/kore_fleets/starter_bots/ts/test/ObservationTest.ts
import { describe } from 'mocha';
import { expect } from 'chai';

import * as fs from 'fs';

import { Observation } from "../kore/Observation";
    
describe('Observation', () =>  {
    it('valid observation works', () => {
        const rawObs = fs.readFileSync('./test/observation.json', 'utf8');
        const ob = new Observation(rawObs);

        expect(ob.player).to.equal(0);
        expect(ob.step).to.equal(16);
        expect(ob.playerFleets.length).to.equal(4);
        expect(ob.playerFleets[0].size).to.equal(0);
        expect(ob.playerShipyards.length).to.equal(4);
        expect(ob.playerShipyards[0].size).to.equal(1);
    })

    it('full observation works', () => {
        const rawObs = fs.readFileSync('./test/fullob.json', 'utf8');
        const ob = new Observation(rawObs);

        expect(ob.player).to.equal(0);
        expect(ob.step).to.equal(200);
        expect(ob.playerHlt.length).to.equal(2);
        expect(ob.playerFleets.length).to.equal(2);
        expect(ob.playerFleets[0].size).to.equal(1);
        expect(ob.playerShipyards.length).to.equal(2);
        expect(ob.playerShipyards[0].size).to.equal(6);
    })
});